packages/modal/src/positioning/ux-modal-positioning.ts

```diff
@@ -0,0 +1,409 @@
+      instance.preferedPlacement = options.placement;
+    }
+    if (options && options.missingSpaceStrategy) {
+      instance.missingSpaceStrategy = options.missingSpaceStrategy;
+    }
+    if (options && options.hiddenClass) {
+      instance.hiddenClass = options.hiddenClass;
+    }
+    if (options && options.offsetX !== undefined) {
+      instance.offsetX = options.offsetX;
+    }
+    if (options && options.offsetY !== undefined) {
+      instance.offsetY = options.offsetY;
+    }
+    if (options && options.constraintMarginX !== undefined) {
+      instance.constraintMarginX = options.constraintMarginX;
+    }
+    if (options && options.constraintMarginY !== undefined) {
+      instance.constraintMarginY = options.constraintMarginY;
+    }
+    if (options && options.constraintElement) {
+      instance.constraintElement = options.constraintElement;
+    } else {
+      instance.constraintElement = element.parentElement || window;
+    }
+    instance.init();
+    return instance;
+  }
+
+  private init() {
+    // When we init the positioning, it might happen that it occurs
+    // a little too early and the element that we must positionned
+    // is not quite ready. We check this with a DOMRect() and if
+    // the width or height are not set we assume we should wait
+    // a little before positioning
+    this.update();
+    const rect = this.element.getBoundingClientRect();
+    if (rect.width && rect.height) {
+      this.update();
+    } else {
+      // We use queueTask here because queueMicroTask
+      // resolves too early in several occasions
+      this.taskQueue.queueTask(() => {
+        this.update();
+      });
+    }
+  }
+
+  public async update() {
+    this.resetElement();
+    this.prepare();
+    // try the prefered placement
+    const mainPlacement = this.getMainPlacement();
+    let hide = false;
+    if (!this.placeMain(mainPlacement, this.missingSpaceStrategy)) {
+      // placeMain returns false if the placement could not be ideally done
+      // with the requested strategy
+      if (this.missingSpaceStrategy === 'flip') {
+        const alternativePlacement = flip[mainPlacement];
+        if (!this.placeMain(alternativePlacement)) {
+          // if flip doesn't work either, then we force the main placement
+          this.placeMain(mainPlacement, 'ignore');
+        }
+      } else if (this.missingSpaceStrategy === 'hide') {
+        hide = true;
+      }
+    }
+
+    this.element.classList.toggle(this.hiddenClass, hide);
+    if (hide) {
+      return;
+    }
+
+    const secondaryPlacement = this.getSecondaryPlacement(mainPlacement);
+    this.placeSecondary(secondaryPlacement);
+  }
+
+  private resetElement() {
+    this.element.style.position = 'absolute';
+    this.element.style.top = '0';
+    this.element.style.left = '0';
+    this.element.style.right = 'auto';
+    this.element.style.bottom = 'auto';
+    this.element.style.width = 'unset';
+    this.element.style.height = 'unset';
```

Comment:
I'm not sure about this, but it feels a bit repetitive. Maybe:
```suggestion
    const style =  this.element.style;
    style.position = 'absolute';
    style.top = '0';
    left = '0';
    right = 'auto';
    bottom = 'auto';
    width = 'unset';
    height = 'unset';
```
Better for minification/perf at runtime, since less property accesses calls. Probably irrelevant to app perf, but pursuing "every bit counts" religiously here won't hurt? I'm not insisting on this though.

Reply:
I have applied this same logic to other places where `this.element.style` is used. I'm event tempted to ref this property as a private `elementStyle` instance property. I did not do it yet.